torch/_inductor/fx_passes/misc_patterns.py

```diff
@@ -69,6 +69,52 @@ def randperm_index_replacement(x, slice_shape):
         scalar_workaround={"slice_shape": 42},
     )
 
+    # Float8 training patterns
+    E4M3_MAX_POS = torch.finfo(torch.float8_e4m3fn).max
+    E5M2_MAX_POS = torch.finfo(torch.float8_e5m2).max
+
+    def amax_with_scaling_pattern(tensor_x_inp, scale_x, IS_BWD):
+        tensor_x = tensor_x_inp.to(torch.float32) * scale_x
+        if IS_BWD:
```

Comment:
instead of using `IS_BWD`, can we just register replacements for both e4m3 and e5m2 variants in both forward and backward?  Motivation: "e4m3 in fwd, e5m2 in bwd" is only one of the possible recipes.

Reply:
ah yes, the difference in `is_bwd` and `not is_bwd` is actually the different data type, I'll just change the name to `is_e5m2` to make it more clear.